src/metabase/models/collection/graph.clj

```diff
@@ -170,13 +170,15 @@
          new-perms          (into {} (for [[group-id collection-id->perms] new-perms]
                                       [group-id (select-keys collection-id->perms (keys (get old-perms group-id)))]))
          [diff-old changes] (data/diff old-perms new-perms)]
-     (data-perms.graph/log-permissions-changes diff-old changes)
```

Comment:
We were logging before checking that the revision numbers were correct. That seems wrong, so I change the order that this happens too.